src/policy/truc_policy.cpp

```diff
@@ -57,13 +59,14 @@ struct ParentInfo {
 
 std::optional<std::string> PackageTRUCChecks(const CTransactionRef& ptx, int64_t vsize,
                                            const Package& package,
-                                           const CTxMemPool::setEntries& mempool_ancestors)
+                                           const CTxMemPool::setEntries& mempool_ancestors,
+                                           const size_t tx_index, const std::vector<std::vector<size_t>>& relations)
 {
     // This function is specialized for these limits, and must be reimplemented if they ever change.
     static_assert(TRUC_ANCESTOR_LIMIT == 2);
     static_assert(TRUC_DESCENDANT_LIMIT == 2);
 
-    const auto in_package_parents{FindInPackageParents(package, ptx)};
+    const auto in_package_parents{FindInPackageParents(tx_index, relations)};
```

Comment:
I think most of the benefits of precomputing this would be realized in the loop a few lines down looking for a package_tx sibling. As is, I'm not sure how much faster precomputation is: n=2 in normal cases, and we don't need to continue computing parents once we've realized it's oversize.